### PR TITLE
fix ctrl-c and refactor itests

### PR DESCRIPTION
**Pull request by adlrocha**
This PR:
- [x] Fixes an issue where itests couldn't be killed with `ctrl-c` 
- [x] It refactor Mir itests so they are run individually and not in batch so we can select specific tests individually instead of having to comment the ones we don't want to run.

### CODE REVIEW

**Review by dnkolegov — APPROVED** [review]
LGTM! Big thanks for fixing the CRLC-C bug!

File: itests/mir_test.go [hunk 1]
```diff
@@ -74,26 +59,88 @@ func runMirManglingTests(t *testing.T, opts ...interface{}) {
 	t.Run("testMirMiningWithMessaging", ts.testMirAllNodesMiningWithMessaging)
 }
 
-func runMirConsensusTests(t *testing.T, opts ...interface{}) {
-	ts := itestsConsensusSuite{opts: opts}
-
+func TestMirOneNodeMining(t *testing.T) {
+	ts := itestsConsensusSuite{opts: []interface{}{kit.ThroughRPC(), kit.MirConsensus()}}
```
Comment (hmoniz) [inline itests/mir_test.go:63]: It looks like this itestsConsensusSuite is (1) immutable and (2) the same for all tests. I think we can easily avoid all the boilerplate in these tests. First, define `ts` as a package-level variable (alternatively, you can use `TestMain` or `init()` to initialize `ts` but it seems unnecessary and will require you to move this file to its own package). Second, get rid of the `t.Run()` calls and just inline the test logic in the `TestMir*` methods directly.

**Review by hmoniz — CHANGES_REQUESTED** [review]

File: itests/mir_test.go (hunk 1 above)
Comment (hmoniz) [reply itests/mir_test.go:63]: In fact, I don't think you need the `iTestsConsensusSuite` struct at all. Just define at package-level:

`var mirTestOpts = []interface{}{kit.ThroughRPC(), kit.MirConsensus()}`

and pass it to the ensemble calls.

Comment (dnkolegov) [reply itests/mir_test.go:63]: testMir* methods are reused in tests with mangler.

Comment (hmoniz) [reply itests/mir_test.go:63]: My suggestion there is to define `TestMir*WithMangler` test methods that wrap the `TestMir*` test functions with setting/unsetting that environment variable. Alternatively, the test functions that we want to run with the mangler environment variable, can be rewritten as a form of table-driven tests.

Comment (dnkolegov) [reply itests/mir_test.go:63]: Ok. I will do that in another PR. Thank you @hmoniz.

**Pull request merged**